quic/s2n-quic-platform/src/io/testing/model.rs

```diff
@@ -8,10 +8,31 @@ use std::{
     borrow::Cow,
     sync::{
         atomic::{AtomicU16, AtomicU64, Ordering},
-        Arc,
+        Arc, Mutex,
     },
 };
 
+#[derive(Clone, Default)]
+pub struct TxRecorder {
+    packets: Arc<Mutex<Vec<Packet>>>
+}
+
+impl TxRecorder {
+    pub fn get_packets(&self) -> Arc<Mutex<Vec<Packet>>> {
+        Arc::clone(&self.packets)
```

Comment:
You can just use `.clone()` here.

```suggestion
        self.packets.clone()
```

Reply:
This was an intentional choice going off of this: https://rust-lang.github.io/rust-clippy/master/#clone_on_ref_ptr

I don't have a strong opinion on this, happy to switch back.

Reply:
Hmm. We don't have that check enabled (it's not by default). Probably best to consider that separately from this.